tests/contrib/forecast/test_forecaster.py

```diff
@@ -0,0 +1,119 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+import torch
+
+import pyro
+import pyro.distributions as dist
+import pyro.poutine as poutine
+from pyro.contrib.forecast import Forecaster, ForecastingModel
+from pyro.infer.reparam import LinearHMMReparam, StableReparam
+
+
+class Model0(ForecastingModel):
+    def model(self, zero_data, covariates):
+        with pyro.plate_stack("batch", zero_data.shape[:-2], rightmost_dim=-2):
+            loc = zero_data[..., :1, 0]
+            scale = pyro.sample("scale", dist.LogNormal(loc, 1))
+
+            with self.time_plate:
+                jumps = pyro.sample("jumps", dist.Normal(0, scale))
+            prediction = jumps.cumsum(-1).unsqueeze(-1) + zero_data
+
+            noise_dist = dist.Laplace(zero_data, 1).to_event(2)
```

Comment:
i guess this `to_event` logic will be tricky for some users...

Reply:
Good point, I can add an adapter inside the `.predict()` method
```py
noise_dist = noise_dist.to_event(2 - niose_dist.event_dim)
```
I'll update shortly...

Reply:
done, and updated tests

Reply:
cool great this automation works